### PR TITLE
[UNR-5334] Allow actor component interest

### PR DESCRIPTION
**Pull request by mironec**
#### Description
Describe your changes here.

#### Release note
REQUIRED: Add a release note to the `##Unreleased` section of CHANGELOG.md. You can find guidance for writing useful release notes [here](../SpatialGDK/Extras/internal-documentation/how-to-write-good-release-notes.md). Documentation changes are exempt from this requirement.

#### Tests
How did you test these changes prior to submitting this pull request?
What automated tests are included in this PR?

STRONGLY SUGGESTED: How can this be verified by QA?

#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?

#### Reminders (IMPORTANT)
If your change relies on a breaking engine change:
* Increment `SPATIAL_ENGINE_VERSION` in `Engine\Source\Runtime\Launch\Resources\SpatialVersion.h` (in the engine fork) as well as `SPATIAL_GDK_VERSION` in `SpatialGDK\Source\SpatialGDK\Public\Utils\EngineVersionCheck.h`. This helps others by providing a more helpful message during compilation to make sure the GDK and the Engine are up to date.

If your change modifies the schema of a snapshot, or the contents of a default snapshot:
* Update `SPATIAL_SNAPSHOT_VERSION_INC` (and optionally `SPATIAL_SNAPSHOT_SCHEMA_HASH`) in `SpatialGDK\Source\SpatialGDK\Public\SpatialConstants.h`. This helps others by providing runtime feedback that snapshots are out of date on servers and/or clients.

If your change updates `Setup.bat`, `Setup.sh`, core SDK version, any C# tools in `SpatialGDK\Build\Programs\Improbable.Unreal.Scripts`, or hand-written schema in `SpatialGDK\Extras\schema`:
* Increment the number in `RequireSetup`. This will automatically run `Setup.bat` or `Setup.sh` when the GDK is next pulled.

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialClassInfoManager.h [hunk 1]
```diff
@@ -114,7 +114,8 @@ class SPATIALGDK_API USpatialClassInfoManager : public UObject
 	ESchemaComponentType GetCategoryByComponentId(Worker_ComponentId ComponentId);
 	const TArray<Schema_FieldId>& GetFieldIdsByComponentId(Worker_ComponentId ComponentId);
 
-	Worker_ComponentId GetComponentIdForClass(const UClass& Class) const;
+	Worker_ComponentId GetComponentIdForActorClass(const UClass& Class) const;
```
Comment (improbable-dmitrii) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialClassInfoManager.h:117]: Hmm, do we want to make this take a `const TSubclassOf<AActor>`?

**Comment by improbable-dmitrii** [comment]
Everything looks good, just needs a changelog entry.

File: SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialClassInfoManager.h (hunk 1 above)
Comment (mironec) [reply SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialClassInfoManager.h:117]: though Unreal likes to throw around random UClass pointers, so may make usage more difficult

Comment (improbable-dmitrii) [reply SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialClassInfoManager.h:117]: TSubclassOf is implicitly constructible from UClass pointer, so this shouldn't change anything from the caller's POV (while defying the point of using it in the first place, as there aren't a lot of runtime checks :/)

**Pull request closed without merging**